daemon/create_test.go

```diff
@@ -0,0 +1,22 @@
+package daemon
+
+import (
+	"testing"
+
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/errdefs"
+	"github.com/stretchr/testify/assert"
+)
+
+// Test case for 35752
+func TestVerifyNetworkingConfig(t *testing.T) {
+	name := "mynet"
+	endpoints := make(map[string]*network.EndpointSettings, 1)
+	endpoints[name] = nil
+	nwConfig := &network.NetworkingConfig{
+		EndpointsConfig: endpoints,
+	}
+	err := verifyNetworkingConfig(nwConfig)
+	assert.True(t, errdefs.IsInvalidParameter(err))
+	assert.EqualError(t, err, "no EndpointSettings for mynet")
```

Comment:
Sorry, I'd prefer to remove this assertion since error messages can change and should not be relied upon.
If you want to check that the cause is a specific type (or something defined at the package level) then that'd be ok.

Reply:
Thanks @cpuguy83. Having a specific error type but loose the restriction on error message itself makes sense in many situations. The PR has been updated.